cirq-core/cirq/transformers/dynamical_decoupling.py

```diff
@@ -98,114 +110,75 @@ def _is_single_qubit_gate_moment(moment: 'cirq.Moment') -> bool:
     return True
 
 
-def _is_clifford_moment(moment: 'cirq.Moment') -> bool:
-    for op in moment.operations:
-        if op.gate is not None and isinstance(op.gate, cirq.MeasurementGate):
-            return False
-        if not cirq.has_stabilizer_effect(op):
-            return False
-    return True
-
-
-def _get_clifford_pieces(circuit: 'cirq.AbstractCircuit') -> list[Tuple[int, int]]:
-    clifford_pieces: list[Tuple[int, int]] = []
-    left = 0
-    for moment_id, moment in enumerate(circuit):
-        if not _is_clifford_moment(moment):
-            clifford_pieces.append((left, moment_id))
-            left = moment_id + 1
-    if left < len(circuit):
-        clifford_pieces.append((left, len(circuit)))
-    return clifford_pieces
+def _is_clifford_op(op: 'cirq.Operation') -> bool:
+    if op.gate is not None and isinstance(op.gate, cirq.MeasurementGate):
+        return False
+    return cirq.has_stabilizer_effect(op)
 
 
 def _is_insertable_moment(moment: 'cirq.Moment', single_qubit_gate_moments_only: bool) -> bool:
     return _is_single_qubit_gate_moment(moment) or not single_qubit_gate_moments_only
 
 
+def _merge_single_qubit_ops_to_phxz(
+    q: 'cirq.Qid', ops: Tuple['cirq.Operation', ...]
+) -> 'cirq.Operation':
+    """Merges [op1, op2, ...] and returns an equivalent op"""
+    if len(ops) == 1:
+        return ops[0]
+    matrices = [cirq.unitary(op) for op in reversed(ops)]
+    product = reduce(np.matmul, matrices)
+    gate: Optional['cirq.Gate'] = single_qubit_decompositions.single_qubit_matrix_to_phxz(product)
+    if gate is None:
+        # gate is close to Identity.
+        gate = cirq.I
+    return gate.on(q)
+
+
+def _merge_single_qubit_ops_of_two_moments(
+    m1: 'cirq.Moment', m2: 'cirq.Moment'
+) -> Tuple['cirq.Moment', ...]:
+    for q in m1.qubits | m2.qubits:
+        op1 = m1.operation_at(q)
+        op2 = m2.operation_at(q)
+        if op1 is not None and op2 is not None:
```

Comment:
if we only care about common qubits then why iterate over `m1.qubits | m2.qubits`? lets iterate over `m1.qubits & m2.qubits`